cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java

```diff
@@ -0,0 +1,256 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+public class KafkaBrokerConfigurationDiffTest {
+
+    private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
+    private static final String KAFKA_VERSION = "2.4.0";
+    KafkaVersion kafkaVersion = VERSIONS.version(KAFKA_VERSION);
+    private int brokerId = 0;
+
+    public ConfigMap getTestingDesiredConfiguration(ArrayList<ConfigEntry> additional) {
+        InputStream is = getClass().getClassLoader().getResourceAsStream("desired-kafka-broker.conf");
+        String desiredConfigString = TestUtils.readResource(is);
+
+        for (ConfigEntry ce: additional) {
+            desiredConfigString += "\n" + ce.name() + "=" + ce.value();
+        }
+
+        ConfigMap configMap = new ConfigMap();
+
+        HashMap<String, String> data = new HashMap();
+        data.put("server.config", desiredConfigString);
+        configMap.setData(data);
+        return configMap;
+    }
+
+    public Map<ConfigResource, Config> getTestingCurrentConfiguration(ArrayList<ConfigEntry> additional) {
+        Map<ConfigResource, Config> current = new HashMap<>();
+        ConfigResource cr = new ConfigResource(ConfigResource.Type.BROKER, "0");
+        InputStream is = getClass().getClassLoader().getResourceAsStream("current-kafka-broker.conf");
+
+        List<String> configList = Arrays.asList(TestUtils.readResource(is).split(System.getProperty("line.separator")));
+        List<ConfigEntry> entryList = new ArrayList<>();
+        configList.forEach(entry -> {
+            String[] split = entry.split("=");
+            String val = split.length == 1 ? "" : split[1];
+            ConfigEntry ce = new ConfigEntry(split[0].replace("\n", ""), val, true, true, false);
+            entryList.add(ce);
+        });
+        for (ConfigEntry ce: additional) {
+            entryList.add(ce);
+        }
+
+        Config config = new Config(entryList);
+        current.put(cr, config);
+        return current;
+    }
+
+    @Test
+    public void testDefaultValue() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(new ArrayList<>()), getTestingDesiredConfiguration(ces), kafkaVersion, brokerId);
+        assertThat(kcd.isDesiredPropertyDefaultValue("offset.metadata.max.bytes"), is(true));
+    }
+
+    @Test
+    public void testNonDefaultValue() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("offset.metadata.max.bytes", "4097", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(ces), getTestingDesiredConfiguration(new ArrayList<>()), kafkaVersion, brokerId);
```

Comment:
For all these tests it would be _so_ much easier to read if:

* You used `singletonList()`, rather than creating a `new ArrayList()`.
* You inlined `ces`.
* You broker the long 3rd line